strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -277,39 +277,263 @@ def reset(self, pure=True, **kwargs):
         self.circuit.reset(self._init_modes)
 
     def prepare_thermal_state(self, nbar, mode):
-        self.circuit.init_thermal(nbar, mode)
+        # self.circuit.init_thermal(nbar, mode)
+        pass
 
     def prepare_vacuum_state(self, mode):
-        self.circuit.loss(0.0, mode)
+        # self.circuit.loss(0.0, mode)
+        pass
 
     def prepare_coherent_state(self, r, phi, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.displace(r, phi, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.displace(r, phi, mode)
+        pass
 
     def prepare_squeezed_state(self, r, phi, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.squeeze(r, phi, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.squeeze(r, phi, mode)
+        pass
 
     def prepare_displaced_squeezed_state(self, r_d, phi_d, r_s, phi_s, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.squeeze(r_s, phi_s, mode)
-        self.circuit.displace(r_d, phi_d, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.squeeze(r_s, phi_s, mode)
+        # self.circuit.displace(r_d, phi_d, mode)
+        pass
 
-    def prepare_cat(self, alpha, phi, desc):
+    def prepare_cat(self, alpha, phi, cutoff, desc="real"):
         """ Prepares the arrays of weights, means and covs for a cat state"""
-        return
+
+        # case alpha = 0 -> prepare vacuum
+        if np.isclose(np.absolute(alpha), 0):
+            return [np.array([1], dtype=float), np.array([0, 0], dtype=float), 0.5 * np.identity(2)]
+
+        norm = 1 / (2 * (1 + np.exp(-2 * np.absolute(alpha) ** 2) * np.cos(phi)))
+        phi = np.pi * phi
+
+        if desc == "complex":
+            rplus = np.sqrt(2 * self.circuit.hbar) * np.array([alpha.real, alpha.imag])
+            cplx_coef = np.exp(-2 * np.absolute(alpha) ** 2 - 1j * phi)
+            rcomplex = np.sqrt(2 * self.circuit.hbar) * np.array(
+                [1j * alpha.imag, -1j * alpha.real]
+            )
+            weights = norm * np.array([1, 1, cplx_coef, np.conjugate(cplx_coef)])
+            weights /= np.sum(weights)
+            means = np.array([rplus, -rplus, rcomplex, np.conjugate(rcomplex)])
+            cov = 0.5 * np.identity(2, dtype=float)
+            cov = np.repeat(cov[None, :], weights.size, axis=0)
+            return weights, means, cov
+
+        elif desc == "real":
```

Comment:
From plotting the Wigner function, I do not find that "real" and "complex" yield the same function.